x/leverage/keeper/liquidate.go

```diff
@@ -0,0 +1,216 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/umee-network/umee/v2/x/leverage/types"
+)
+
+// getLiquidationAmounts takes a repayment and reward denom proposed by a liquidator and calculates
+// the actual repayment amount a target address is eligible for, and the corresponding collateral
+// to burn and rewards to return to the liquidator.
+func (k Keeper) getLiquidationAmounts(
+	ctx sdk.Context,
+	liquidatorAddr sdk.AccAddress,
+	targetAddr sdk.AccAddress,
+	maxRepay sdk.Coin,
+	rewardDenom string,
+) (tokenRepay sdk.Coin, collateralBurn sdk.Coin, tokenReward sdk.Coin, err error) {
+	repayDenom := maxRepay.Denom
+	collateralDenom := k.FromTokenToUTokenDenom(ctx, rewardDenom)
+
+	// get relevant liquidator, borrower, and module balances
+	borrowerCollateral := k.GetBorrowerCollateral(ctx, targetAddr)
+	totalBorrowed := k.GetBorrowerBorrows(ctx, targetAddr)
+	availableRepay := k.bankKeeper.SpendableCoins(ctx, liquidatorAddr).AmountOf(repayDenom)
+
+	// calculate borrower health in USD values
```

Comment:
```suggestion
	// calculate total borrowed value in USD
```

Reply:
keeping as-is (the comment describes the next three steps: `borrowed value`, `liquidation threshold`, and check liquidation eligibility - which together comprise the borrower's health in this context)

Reply:
Personally I found it a bit confusing, because we don't have a clear health metric (eg, I would imagine health as a number between 0-1, 1=perfect). What we really calculate is the total borrowed value in USD.